JSON.cpp

```diff
@@ -0,0 +1,114 @@
+                                continue;
+                            }
+                            pos1 = val.find_first_not_of(" \t\v");
+                            if (pos1 != std::string::npos && val[pos1] == '\"'){
+                                val = val.substr(pos1 + 1);
+                                pos1 = val.find("\"");
+                                listvalues.push_back(val.substr(0, pos1));
+                            }
+                            val = val.substr(pos1);
+                            pos1 = val.find_first_not_of(" \t\v\"");
+                            if (pos1 != std::string::npos && val[val.find_first_not_of(" \t\v\"")] == ']') break;
+                            std::getline(f, val);
+                            //std::cout << val << "\neddig ok [ val\n";
+                        } while (pos1 == std::string::npos || val[val.find_first_not_of(" \t\v")] != ']');
+                        //std::cout << val << "\neddig ok [ val outside \n";
+                        t = "";
+                    }else{
+                        val = val.substr(0,val.find_first_of(","));
+                        if (val.find('.') != std::string::npos){
+                            vartype = doubl;
+                        }else{
+                            vartype = doubl;
+                        }
+                        t = t.substr(t.find(val) + val.size());
+                        t = t.substr(t.find_first_of("\n,") + 1);
+                    }
+                    if (t.find_first_not_of(" ,\n\t\"") == std::string::npos || t[t.find_first_not_of(",\n\t\" ")] == '}'){
+                        t = "";
+                    }
+                    //std::cout << val << "\neddig map ins begin\n";
+                    std::variant<std::string, int, double, list> mapvalue;
+                    if (vartype == string)
+                    {
+                        mapvalue = val;
+                    }
+                    else if (vartype == integer)
+                    {
+                        mapvalue = std::stoi(val);
+                    }
+                    else if (vartype == doubl)
+                    {
+                        mapvalue = std::stod(val);
+                    }
+                    else if (vartype == ls)
+                    {
+                        //std::cout << t << "\neddig map ins listval\n";
+                        mapvalue = list(listvalues);
+                    }
+                    parsed[p1] = mapvalue;
+                }
+            }
+        }catch(const int& e){
+            switch (e) {
+            case -1:
+                std::cerr << "double declaration of variable name\n";
+                std::exit(-1);
+            case -2:
+                //std::cerr << "no key found";
+                continue;
+            case -3:
+                std::cerr << "Invalid json syntax\n";
+                std::exit(-1);
+            }
```

Comment:
Ezt azért vesézzük ki egy kicsit. Ha már van exception header-ünk, nyugodtan dobálhatnánk kivételeket is, de ez csak egy megjegyzés. 
Összetett típusok esetén kell referencia szerint elkapni, inteknél olyan túl sok értelme nincsen.
Ha az érzékeim nem csalnak akkor a `case -2:` re semmi szükséged nincsen. Ehhez kapcsolódó bónusz tipp: Kódszemét!